### PR TITLE
Add executed states to field for validating execution path in tests.

### PR DESCRIPTION
**Pull request by LiamRiddell**
See #28

### CODE REVIEW

**Comment by LiamRiddell** [comment]
Need to fix my fork one second

**Comment by LiamRiddell** [comment]
@oshikiri if you could review, please.

**Comment by LiamRiddell** [comment]
We should squash and merge these commits so commits doesn't get repeated.

File: test/FakeStateMachine-run.ts [hunk 1]
```diff
@@ -123,4 +124,60 @@ describe('FakeStateMachine#run()', () => {
...
+          true
+        )
+      );
+    });
+  });
+
+  describe('should have execution path', () => {
+    const definition = require('./fixtures/definitions/execution-path-multiple.json');
+
+    test('execution path should be valid', async () => {
```
Comment (oshikiri) [inline test/FakeStateMachine-run.ts:155]: "valid" is ambiguous, so could you use more detailed description? For example, "executionPath has names in executed states" (I have no confidence about english)

File: test/FakeStateMachine-run.ts [hunk 2]
```diff
@@ -123,4 +124,60 @@ describe('FakeStateMachine#run()', () => {
...
+        'Success',
+      ]);
+    });
+
+    test('execution path should be invalid', async () => {
+      const fakeStateMachine = new FakeStateMachine(definition, {});
+
+      await fakeStateMachine.run({});
+
+      expect(fakeStateMachine.executionPath).not.toEqual([
```
Comment (oshikiri) [inline test/FakeStateMachine-run.ts:174]: Could you describe an expected behaviour? I think `fakeStateMachine.executionPath` should be empty list, wdyt?

File: test/FakeStateMachine-run.ts [hunk 3]
```diff
@@ -123,4 +124,60 @@ describe('FakeStateMachine#run()', () => {
       );
     });
   });
+
+  describe('when Pass state has undefined ResultPath', () => {
```
Comment (oshikiri) [inline test/FakeStateMachine-run.ts:128]: Is it required?

**Review by oshikiri — CHANGES_REQUESTED** [review]
Thank you for contributing!

I wrote some comments. Please check it

File: test/FakeStateMachine-run.ts (hunk 1 above)
Comment (LiamRiddell) [reply test/FakeStateMachine-run.ts:155]: No problem, I'll see if I can make it clearer.

File: test/FakeStateMachine-run.ts (hunk 2 above)
Comment (LiamRiddell) [reply test/FakeStateMachine-run.ts:174]: This is validating that the `executionPath` does not contain evaluated rules for choice states. The exected behavior should only have executed states in the order of execution. I'm checking this by making sure that the state `NonExecutingRule` has not been added to the `executionPath`.

File: test/FakeStateMachine-run.ts (hunk 3 above)
Comment (LiamRiddell) [reply test/FakeStateMachine-run.ts]: Not sure what happened here it looks like it's a side effect of #25. I've removed.

File: test/FakeStateMachine-run.ts (hunk 2 above)
Comment (oshikiri) [reply test/FakeStateMachine-run.ts:150]: I still don't fully understand

The assertion at L136 seems to be more severe, so we can simply remove this test case, can't we?

Comment (LiamRiddell) [reply test/FakeStateMachine-run.ts:150]: Sure, I'll remove it now.

**Review by oshikiri — APPROVED** [review]

**Comment by oshikiri** [comment]
Released as v0.6.0

**Pull request merged**